### DISCUSSION TITLE
建议正向 WebSocket 响应头添加 Sec-WebSocket-Protocol

### DISCUSSION THREAD

**Discussion by he0119** [howmanybots/onebot#232] [想法 / 建议]
这样可以方便应用端获取实现端的标准版本与实现名。同时也与反向 WebSocket 标准统一。

**Comment by he0119** [comment]
>| Sec-WebSocket-Protocol |, 可选, 客户端发起握手的时候可以在头部设置该字段, 该字段的值是一系列客户端希望在于服务端交互时使用的子协议 (subprotocol), 多个子协议之间用逗号分隔, 按客户端期望的顺序降序排列, 服务端可以根据客户端提供的子协议列表选择一个或多个子协议

deno 响应的协议必须在客户端请求的协议之中，否则会

```text
TypeError: Protocol '12.test' not in the request's protocol list (non negotiable)
```

**Comment by he0119** [comment]
RC | OneBot 2022/11/15 23:18:35
connect元事件
RC | OneBot 2022/11/15 23:18:42
不可避免了

**Comment by stdrc** [comment]
#233 添加了 `connect` 元事件，另外，反向 WS 的 `Sec-WebSocket-Protocol` 暂时保持现状

**Reply by Super-Lollipop** [reply]
meta.connect这个事件内容应该可以放在websocket握手过程的header吧

**Reply by stdrc** [reply]
不是很想让 OneBot 的“业务逻辑”相关事件跟“OneBot Connect”的连接行为深度耦合
